erdpy_wallet/user_keys.py

```diff
@@ -43,7 +48,7 @@ def __init__(self, buffer: bytes) -> None:
         if len(buffer) != USER_PUBKEY_LENGTH:
             raise ErrBadPublicKeyLength()
 
-        self.buffer = buffer
+        self.buffer = bytes(buffer)
```

Comment:
I don't think the cast to bytes does anything since the `buffer` variable it's already of type bytes.

Reply:
Mypy / Pyright will not complain about this code (passing `bytearray` instead of `bytes`):

```
buffer_hex = "..."
buffer = bytes.fromhex(buffer_hex)
secret_key = UserSecretKey(bytearray(buffer))
```

However, if secret_key is initialized using a bytearray, `nacl.signing.SigningKey(buffer)` fails, due to (`nacl` code):

```
if not isinstance(seed, bytes):
    raise exc.TypeError(
    "SigningKey must be created from a 32 byte seed"
)
```

Thus, we've added `bytes(buffer)` to fix misuse (passing `bytearray` instead of `bytes`).